cylc print: always descend at top level

matthewrmshin opened the pull request:
Allow listing to descend at the top level, regardless of items in the directory.

hjoliver commented on lib/cylc/suite_srv_files_mgr.py:
```diff
@@ -328,8 +328,10 @@ def list_suites(self, regfilter=None):
         skip_names = [
             "log", "share", "work", self.DIR_BASE_SRV, self.FILE_BASE_SUITE_RC]
         for dirpath, dnames, fnames in os.walk(run_d, followlinks=True):
-            # Don't descent further if it looks like a suite directory
```
"decend" (with a 'd' on the end)?

matthewrmshin replied:
Comment fixed.

matthewrmshin commented:
Branch updated, squashed and re-based.

hjoliver commented:
To be more specific, we can now have a directory hierarchy of installed suites, e.g. `cylc reg nwp/one foo` and `cylc reg nwp/two` results in:
```
$ cylc print nwp
nwp/two | No title provided | ~/foo
nwp/one | No title provided | ~/bar
```
Then (on this branch) `mkdir ~/cylc-run/nwp/work` results in:
```
$ cylc print
# (nothing!)
```

Maybe presence of a `.service` sub-dir would be more definitive?

matthewrmshin commented:
> Maybe presence of a `.service` sub-dir would be more definitive?

Yes, but this logic exists to prevent `os.walk` from descending into the sub-directories of a cylc-6 suite, which will not have a `.service`. I'll have another think to see if there is a better way. Maybe the safest way is to look for something like `cylc-suite.db`. (You have to be asking for trouble if you create a directory called `cylc-suite.db`!)

hjoliver commented:
> Yes, but this logic exists to prevent os.walk from descending into the sub-directories of a cylc-6 suite, 

Ah, OK, I hadn't quite understood that.

> You have to be asking for trouble if you create a directory called cylc-suite.db!

Agreed, that seems safe.

matthewrmshin commented:
Branch updated, squashed and re-based.

I have now modified the logic to skip only if:
* The current directory is not the top directory.
* A sub-directory is `.service/`.
* A file in the current directory is `cylc-suite.db`.
